client/my-sites/comments/comment/comment-html-editor.jsx

```diff
@@ -0,0 +1,235 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { localize } from 'i18n-calypso';
+import { delay, map, reduce } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import AddImageDialog from 'post-editor/editor-html-toolbar/add-image-dialog';
+import AddLinkDialog from 'post-editor/editor-html-toolbar/add-link-dialog';
+import {
+	closeHtmlTag,
+	insertCustomContent,
+	insertHtmlTag,
+	insertHtmlTagOpenClose,
+	insertHtmlTagSelfClosed,
+	insertHtmlTagWithText,
+	setCursorPosition,
+	splitSelectedContent,
+} from 'lib/html-toolbar';
+
+export class CommentHtmlEditor extends Component {
+	static propTypes = {
+		commentContent: PropTypes.string,
+		disabled: PropTypes.bool,
+		onChange: PropTypes.func,
+	};
+
+	state = {
+		openTags: [],
+		selectedText: '',
+		showImageDialog: false,
+		showLinkDialog: false,
+	};
+
+	storeTextareaRef = textarea => ( this.textarea = textarea );
+
+	isTagOpen = tag => -1 !== this.state.openTags.indexOf( tag );
+
+	insertHtmlTag = tag => {
+		const isTagOpen = insertHtmlTag(
+			this.textarea,
+			tag,
+			this.isTagOpen( tag.name ),
+			this.setCommentContentValue
+		);
+		this.setState( ( { openTags } ) => ( {
+			openTags: isTagOpen
+				? openTags.concat( tag.name )
+				: openTags.filter( openTag => openTag !== tag.name ),
+		} ) );
+	};
+
+	onClickBold = () => this.insertHtmlTag( { name: 'strong' } );
```

Comment:
Minor note, but naming wise, we prefer more descriptive names. Say `insertStrongTag` over `onClick` or `insertClosingTags` over `onClickCloseTags`

Reply:
I wanted to keep the naming as close to the post editor HTML toolbar (which, admittedly, are my bad as well 😅).
Though, as we move farther and farther from the post editor, I don't see any reason to keep this consistency in place. 🙂

Reply:
Whatever feels best to you. As I mentioned before it's okay for us to diverge here, since I think we'll come up with different needs.